src/modules/sharing/sharing.controller.ts

```diff
@@ -71,11 +73,67 @@ export class SharingController {
   async getPublicSharing(
     @Param('sharingId') sharingId: Sharing['id'],
     @Query('code') code: string,
+    @Headers('x-share-password') password: string | null,
   ) {
     if (!code) {
       throw new BadRequestException('Code is required');
     }
-    return this.sharingService.getPublicSharingById(sharingId, code);
+    return this.sharingService.getPublicSharingById(sharingId, code, password);
+  }
+
+  @Get('/public/:sharingId/item')
+  @Public()
+  @ApiOperation({
+    summary: 'Get sharing item info',
+  })
+  @ApiParam({
+    name: 'sharingId',
+    description: 'Id of the sharing',
+    type: String,
+  })
+  @ApiOkResponse({ description: 'Get sharing item info' })
+  async getPublicSharingItemInfo(@Param('sharingId') sharingId: Sharing['id']) {
+    return this.sharingService.getPublicSharingItemInfo(sharingId);
+  }
+
+  @Post('/:sharingId/password')
+  @ApiOperation({
+    summary: 'Set password for public sharing',
+  })
+  @ApiParam({
+    name: 'sharingId',
+    description: 'Id of the sharing',
+    type: String,
+  })
+  @ApiOkResponse({ description: 'Sets/edit password for public sharings' })
+  async setPublicSharingPassword(
+    @UserDecorator() user: User,
+    @Param('sharingId') sharingId: Sharing['id'],
+    @Body() sharingPasswordDto: SetSharingPasswordDto,
+  ) {
+    const { encryptedPassword } = sharingPasswordDto;
```

Comment:
Could we somewhat validate the format of this? Is it not an empty string, has the expected format (hex or whatever, etc)?

Reply:
The Dto already throws an error if the string is empty with  decorator @IsNotEmpty(). 

The password is limited to 50 characters in the frontend, but we receive it already encrypted so we can not check this.

However, I see `@internxt/lib` encodes the returned encryptedText into base64, so I added the @IsBase64() check to `set-sharing-password.dto.ts`. What do you think?